### PR TITLE
Add a helm chart version with some backed rules

### PR DESCRIPTION
**Pull request by pipo02mix**
Start using helm package manager to maintain the resources and update components. Add some basic rules to Prometheus by default.

Towards https://github.com/giantswarm/giantswarm/issues/4534

### CODE REVIEW

File: helm/README.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
+# Prometheus
+
+Here it is a new opinionated [Prometheus](https://prometheus.io/) recipe thought to be deployed in the Giant Swarm tenant clusters.
```
Comment (JosephSalisbury) [inline helm/README.md]:
```suggestion
An opinionated [Prometheus](https://prometheus.io/) Helm chart for deployment in Giant Swarm clusters.
```

File: helm/README.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+## Prerequisites
+
+- Kubernetes 1.3+ with Beta APIs enabled
```
Comment (JosephSalisbury) [inline helm/README.md]: Beta APIs?

File: helm/README.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+```
+
+The command deploys Prometheus on the Kubernetes cluster in the default configuration. The [configuration](#configuration) section lists the parameters that can be configured during installation.
+
+## Uninstalling the Chart
+
+To uninstall/delete the deployment:
+
+```console
+$ helm delete prometheus
```
Comment (JosephSalisbury) [inline helm/README.md]: Do we want `--purge` here as well?

File: helm/README.md (hunk 2 above)
Comment (pipo02mix) [reply helm/README.md]: Copied from the original stable chart. I guess at that time you needed beta api gates

**Comment by pipo02mix** [comment]
Yeah, it is huge, I can split it. Indeed it is the stable chart from the community but I made some arrangements in order to work against our rules (I added also some of our rules). I have removed the exporters because they are already in the tenant clusters running, so we don't copy configuration (ex. `kubernetes-node-exporter` repo already has this configuration)

File: helm/README.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
+# Prometheus
+
+An opinionated [Prometheus](https://prometheus.io/) Helm chart for deployment in Giant Swarm clusters.
```
Comment (puja108) [inline helm/README.md]: Can we add that this is based off of upstream stable chart or so, we should give credit

File: helm/README.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+`serverFiles.alerts` | Prometheus server alerts configuration | `{}`
+`serverFiles.rules` | Prometheus server rules configuration | `{}`
+`serverFiles.prometheus.yml` | Prometheus server scrape configuration | example configuration
+`extraScrapeConfigs` | Prometheus server additional scrape configuration | ""
+`networkPolicy.enabled` | Enable NetworkPolicy | `false` |
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```console
+$ helm install stable/prometheus --name prometheus \
```
Comment (puja108) [inline helm/README.md]:
```suggestion
$ helm install ./helm --name prometheus \
```

File: helm/README.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    --set server.terminationGracePeriodSeconds=360
+```
+
+Alternatively, a YAML file that specifies the values for the above parameters can be provided while installing the chart. For example,
+
+```console
+$ helm install stable/prometheus --name prometheus -f values.yaml
```
Comment (puja108) [inline helm/README.md]:
```suggestion
$ helm install ./helm --name prometheus -f values.yaml
```

File: helm/README.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+All inbound connections to Prometheus Server are still allowed.
+
+To enable network policy for Prometheus, install a networking plugin that
+implements the Kubernetes NetworkPolicy spec, and set `networkPolicy.enabled` to true.
+
+If NetworkPolicy is enabled for Prometheus' scrape targets, you may also need
+to manually create a networkpolicy which allows it.
+
+
+__Note__: This chart configuration is highly influences by the [community one](https://github.com/helm/charts/tree/master/stable/prometheus).
```
Comment (puja108) [inline helm/README.md]:
```suggestion
__Note__: This chart is based off of the [upstream community chart](https://github.com/helm/charts/tree/master/stable/prometheus).
```

File: helm/README.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+
+- Kubernetes 1.3+ with Beta APIs enabled
+- [Helm](https://helm.sh) package manager.
+
+## Installing the Chart
+
+To install the chart:
+
+```console
+$ helm install ./helm --name prometheus
```
Comment (puja108) [inline helm/README.md]: should we change `prometheus` to `my-prometheus` to show that you can use your own naming here?

File: helm/README.md (hunk 4 above)
Comment (pipo02mix) [reply helm/README.md]: done in the bottom

File: helm/README.md (hunk 8 above)
Comment (pipo02mix) [reply helm/README.md]: good idea

Comment (pipo02mix) [inline helm/README.md]:
```suggestion
$ helm install ./helm --name my-prometheus
```

File: helm/README.md [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+```
+
+The command deploys Prometheus on the Kubernetes cluster in the default configuration. The [configuration](#configuration) section lists the parameters that can be configured during installation.
+
+## Uninstalling the Chart
+
+To uninstall/delete the deployment:
+
+```console
+$ helm delete prometheus --purge
```
Comment (pipo02mix) [inline helm/README.md]:
```suggestion
$ helm delete my-prometheus --purge
```

File: helm/README.md [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+`serverFiles.alerts` | Prometheus server alerts configuration | `{}`
+`serverFiles.rules` | Prometheus server rules configuration | `{}`
+`serverFiles.prometheus.yml` | Prometheus server scrape configuration | example configuration
+`extraScrapeConfigs` | Prometheus server additional scrape configuration | ""
+`networkPolicy.enabled` | Enable NetworkPolicy | `false` |
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```console
+$ helm install ./helm --name prometheus \
```
Comment (pipo02mix) [inline helm/README.md]:
```suggestion
$ helm install ./helm --name my-prometheus \
```

**Comment by pipo02mix** [comment]
Comments addressed PTAL

**Pull request merged**